src/wasm/flb_wasm.c

```diff
@@ -139,6 +170,14 @@ struct flb_wasm *flb_wasm_instantiate(struct flb_config *config, const char *was
         goto error;
     }
 
+#ifdef FLB_WASI_SOCKETS
+    // the address pool must be set before running wasm_runtime_set_wasi_args_ex
+    // otherwise it will be ignored.
+    if (address_pool) {
+        set_address_pool(fw, module, address_pool);
+    }
+#endif
```

Comment:
Looks good for me with the current implementation.

Off topic: I was stuck for not working local address (127.0.0.1) and without `-DFLB_WASI_SOCKETS=On` on cmake configure.
It will be nicer if that option is not enabled and disabled WASI socket feature.

Reply:
I added a default for the address pool as well as refactored the in_exec_wasi code to emit an error when using the address_pool setting when WASI sockets are disabled.